pkg/apis/triggers/v1alpha1/trigger_template_validation.go

```diff
@@ -64,3 +69,29 @@ func validateResourceTemplates(templates []TriggerResourceTemplate) *apis.FieldE
 	}
 	return nil
 }
+
+// Verify every param in the ResourceTemplates is declared with a ParamSpec
+func verifyParamDeclarations(params []pipelinev1.ParamSpec, templates []TriggerResourceTemplate) *apis.FieldError {
+	declaredParamNames := map[string]struct{}{}
+	for _, param := range params {
+		declaredParamNames[param.Name] = struct{}{}
+	}
+	for i, template := range templates {
+		// Get all params in the template $(params.NAME)
+		paramsRegexp := regexp.MustCompile(`\$\(params.(?P<var>[_a-zA-Z][_a-zA-Z0-9.-]*)\)`)
```

Comment:
Can this regex declaration be moved up outside the for loop?

Reply:
Definitely, good catch! 🙂